docs/iam-usage.rst

```diff
@@ -0,0 +1,144 @@
+   for that, please check out `the docs <https://cloud.google.com/iam/docs/>`_.
+
+Resources that implement the IAM interface provide the following methods:
+
+Policy Editing Methods
+----------------------
+
+.. note:: All policy editing methods except ``set_policy`` have an optional
+   ``retries`` keyword argument that takes the number of times to retry in the
+   even of an interrupted transaction
+
+>>> resource.add_roles(iam.user('alice@example.com'), iam.roles.OWNER.name, iam.roles.EDITOR.name)
+True
+
+The first argument is one of the IAM "Member" types:
+
+- ``iam.user(email)`` an individual Google account
+- ``iam.service_account(email)`` a Google Cloud Service Account
+- ``iam.group(email)`` a Google group.
+- ``iam.domain(domain_name)`` a Google-For-Work domain
+- ``iam.ALL_AUTHENTICATED_USERS`` any authenticated user
+- ``iam.ALL_USERS``
+
+Note that all of these are convenience wrappers around strings. See the list of member string formats `here <https://cloud.google.com/iam/docs/managing-policies>`_.
+
+All following arguments are roles to added to the specified member. Roles are simply strings that correspond to sets of permissions.
+The ``iam`` module provides a (possibly incomplete) list of curated roles.
+You can acquire the complete list by running ``gcloud iam list-grantable-roles //fully/qualified/resource/path``, and use the
+names provided there in place of ``iam.roles.XXXX.name`` wherever the latter is used in this doc.
+
+This method returns ``True`` if the change to the resource's policy was made successfully, and returns ``False`` otherwise
+(such as in the case of an update being made during this update).
+
+>>> resource.remove_roles(iam.group('devs@example.com'), iam.roles.OWNER.name, iam.roles.EDITOR.name)
+False
+
+Same as above, but removes the specified roles from the specified member.
+
+>>> resource.add_members(iam.roles.OWNER.name, iam.domain('example.com'), iam.service_account('compute@iam.my-project.example.com'))
+True
+
+Same as above, but the first argument is a role, and all subsequent arguments are members to which that role should be added.
+
+>>> resource.remove_members(iam.roles.OWNER.name, iam.ALL_USERS)
+True
+
+Same as above, but removes all listed members from the specified role.
+
+>>> def remove_fn(member):
+>>>     return iam.is_group(member) or member == iam.user('bob@example.com')
+>>> policy_change = iam.PolicyChange(version=2).add(iam.roles.OWNER.name, iam.user('alice@example.com'))
+>>> policy_change.remove(iam.roles.EDITOR.name, iam.domain('example.com'), iam.group('devs@example.com'))
+>>>              .remove_fn(iam.roles.READER.name, remove_fn)
+>>> policy_change.apply(resource)
+True
+
+If you need more complex logic to modify an IAM policy you can create a ``iam.PolicyChange`` object. ``PolicyChange`` takes
+an optional ``version`` keyword argument which specifies which version the updated policy should be considered.
+
```

Comment:
It seems strange that version is specified cilent-side, this does seem to be how the API works, but if you specify the same version is it an override or an error?